app/controllers/pipeline_work_in_progress_controller.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+class PipelineWorkInProgressController < ApplicationController
```

Comment:
This is just a personal choice but I would generally opt for a service object to do this kind of processing. For various reasons:
- it keeps the controller actions separate from any data processing which makes it easier to debug i.e. if your controller tests fail you will know they are failing because of the request or response.
- it makes the whole thing a bit cleaner and gives you better separation of concerns.
- it makes things easier to test i.e. you can test your controller actions separately from your processing actions.

This is a strategy that I generally use as shown here https://github.com/sanger/traction-service/blob/f36471dc4cfc5e9ec5dcb2bce84c27d1a38079be/app/controllers/v1/pacbio/plates_controller.rb. But again it is personal choice. There is a blog post about it here https://thoughtbot.com/blog/skinny-controllers-skinny-models and this is the approach I generally take.

Reply:
I moved the biggest method into PipelineList as it is generic and seems to belong OK there. I think the others are probably OK here for now.